MapboxCoreNavigation/RouteController.swift

```diff
@@ -687,14 +687,9 @@ extension RouteController: CLLocationManagerDelegate {
     }
     
     func hasEnteredTunnel(_ intersectionBounds: [IntersectionBounds], coordinates: [CLLocationCoordinate2D]) -> Bool {
-        guard let startLocation = coordinates.first else { return false }
-        
         let distanceTraveled = routeProgress.distanceTraveled
         for bounds in intersectionBounds {
-            let distanceToTunnelEntrance = Polyline(coordinates).distance(from: startLocation, to: bounds.entry.location)
-            let distanceToTunnelExit = Polyline(coordinates).distance(from: startLocation, to: bounds.exit.location)
-            
-            if distanceToTunnelEntrance <= distanceTraveled && distanceTraveled <= distanceToTunnelExit {
+            if bounds.distanceToEntry <= distanceTraveled && distanceTraveled <= bounds.distanceToExit {
```

Comment:
This doesn’t address the performance concerns in https://github.com/mapbox/mapbox-navigation-ios/pull/1127#discussion_r170408923. `Polyline.distance(from:to:)` still gets called twice for every pair of intersections along the step on every location update.

Reply:
@1ec5 Indeed, currently looking into it.